radixdlt-engine/src/main/java/com/radixdlt/atommodel/system/construction/NextEpochConstructorV3.java

```diff
@@ -251,6 +254,6 @@ public void construct(SystemNextEpoch action, TxBuilder txBuilder) throws TxBuil
 		validatorKeys.forEach(k -> txBuilder.up(new ValidatorBFTData(k, 0)));
 
 		txBuilder.up(new EpochData(prevEpoch.getEpoch() + 1));
-		txBuilder.up(new RoundData(0, 0));
+		txBuilder.up(new RoundData(0, timestamp));
```

Comment:
Shouldn't this `timestamp` be provided by the action?

Reply:
I don't think so. This is a "system" time, user has no control over it.